src/cfg.rs

```diff
@@ -0,0 +1,120 @@
+use pwd::Passwd;
+use serde::{Deserialize, Serialize};
+use std::env;
+use std::fs::File;
+use std::collections::BTreeMap;
+
+const CONFIG_ENV: &str = "DOORMAN_CONFIG";
+const DEFAULT_CONFIG: &str = "/etc/doorman.yml";
+
+#[derive(Deserialize, Debug)]
+pub struct Options {
+    pub user: String,
+    pub dosemu_container: String,
+    pub rundir: String,
+    pub sysops: Vec<String>,
+}
+
+fn default_door_nodes() -> i8 {
+    return 1;
+}
+
+#[derive(Deserialize, Debug)]
+pub struct Door {
+    pub path: String,
+    #[serde(default = "default_door_nodes")]
+    pub nodes: i8,
+    pub launch: String,
+    pub configure: Option<String>,
+    pub nightly: Option<String>,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct Config {
+    pub doorman: Options,
+    pub doors: BTreeMap<String, Door>,
+}
+
+#[derive(Serialize, Debug)]
+pub struct User {
+    pub uid: u32,
+    pub username: String,
+    pub display_name: String,
+    pub is_sysop: bool,
+}
+
+impl Config {
+    pub fn config_path() -> String {
+        let envvar: Option<String> = match env::var_os(CONFIG_ENV) {
+            Some(val) => match val.to_str() {
+                Some(val_to_str) => Some(String::from(val_to_str)),
+                None => None,
+            },
+            None => None,
+        };
+
+        return String::from(match envvar {
+            Some(val) => val,
+            None => String::from(DEFAULT_CONFIG),
+        });
+    }
```

Comment:
```suggestion
env::var_os(CONFIG_ENV)
            .map(|env| env.into_string().unwrap())
            .unwrap_or(DEFAULT_CONFIG.to_string())
            .to_string()
```

I'm not quite sure what you're trying to do here, it seems to me you want to return either the user set `CONFIG_ENV` or the `DEFAULT_CONFIG` env as String.